src/configuration.ts

```diff
@@ -1278,6 +1295,15 @@ export async function initFromStateAndSettings(): Promise<void> {
                 updatedSettingsSubkeys.push(subKey);
             }
 
+            subKey = "enableLocalDebugRun";
+            let updatedEnableLocalDebugRun : boolean | undefined = workspaceConfiguration.get<boolean>(subKey);
+            if (updatedEnableLocalDebugRun !== enableLocalDebugRun) {
+                readEnableLocalDebugRun();
+                readCurrentLaunchConfiguration();
+                extension.getState().configureDirty = true;
+                updatedSettingsSubkeys.push(subKey);
+            }
+
             // Final updates in some constructs that depend on more than one of the above settings.
```

Comment:
I may be wrong, but I think you don't need configureDirty to be true here. Toggling between view or hide the UI elements doesn't need the extension to go through the reconfigure phase. In case you encountered issues which inspired you to write this (as opposed to looking at other similar code above which may have other reasons) let me know. It doesn't hurt but sometimes, for large projects, configure may be very long and we don't want to let it happen without being necessary.

Same question for calling readCurrentLaunchConfiguration. This doesn't take long, even one less reason to insist on removing it, but do you remember whether you needed that call?

Reply:
>>as opposed to looking at other similar code above which may have other reasons)

Yes, basically I followed the pattern without understanding the implications. I will update the PR using your input.
Thanks a lot for the feedback.